File: code/download_cci_data/get_fire_cci.R
library(RCurl)

# Define the local directory name to put data in
ddir <- "C:/data/3_fire_data/burned_area/fire_cci"

HOSTNAME <- 'ftp.ceda.ac.uk'

USERNAME <- 'nrietze'
PASSWORD <- 'FNrMg^e5Gn5M'

# loop through years
for (year in 2001:2020) {
  
  # create a new directory for the year's data
  ydir <- file.path(ddir, as.character(year))
  
  if (dir.exists(ydir)){
    print('Folder exists. Interrupting download.')
    next
  }
  
  dir.create(ydir, showWarnings = FALSE)
  
  f.URL <- paste0("ftp://", USERNAME, ":", PASSWORD, "@", HOSTNAME,"/neodc/esacci/fire/data/burned_area/MODIS/pixel/v5.1/compressed/", year, "/")
  
  # get the remote files to the local directory
  filenames <- getURL(url = f.URL, 
                      # userpwd = paste(USERNAME, PASSWORD, sep = ":"),
                      ftp.use.epsv = FALSE, dirlistonly = TRUE)
  filenames <- strsplit(filenames, "\r*\n")[[1]]
  
  # loop through months
  for (month in 4:11) {
    print('Processing: ')
    
    cat("Downloading data from:", month, year, "\n")
    
    # define the remote file name pattern
    pattern <- sprintf("%d%02d01", year, month)
    filenames_filtered <- filenames[grepl(pattern, filenames) & grepl("AREA_[134]", filenames)]
    
    for (filename in filenames_filtered) {
      local_filename <- file.path(ydir, filename)
      download.file(paste0(f.URL, "/", filename), 
                    destfile = local_filename, 
                    mode = "wb",
                    quiet = TRUE)
      untar(local_filename, exdir = ydir)
      file.remove(local_filename)
    }
  }
  
  Sys.sleep(10)
}

File: code/download_cci_data/get_fire_cci.py
#!/usr/bin/env python

# Import required python modules
import ftplib
import os
import tarfile

# Define the local directory name to put data in
ddir=r"C:\data\3_fire_data\burned_area\fire_cci"

# Change the local directory to where you want to put the data
os.chdir(ddir)

# login to FTP
f=ftplib.FTP("ftp.ceda.ac.uk", "nrietze", "FNrMg^e5Gn5M")

# loop through years
for year in range(2019,2020):
    
    # create a new directory for the year's data
    ydir = os.path.join(ddir, str(year))
    os.makedirs(ydir, exist_ok=True)
    
    # loop through months
    for month in range(4, 5):
        # change the remote directory
        f.cwd("/neodc/esacci/fire/data/burned_area/MODIS/pixel/v5.1/compressed/%.4d" % year )
        
        # get the remote files to the local directory
        filenames = f.nlst()
        for filename in filenames:
            if filename.endswith(".tar.gz") and ("AREA_1" in filename or "AREA_3" in filename or "AREA_4" in filename) and filename.startswith("%d%.2d" % (year, month)):
                f.retrbinary("RETR %s" % filename, open(filename, "wb").write)
                with tarfile.open(filename, "r:gz") as tar:
                    tar.extractall(path = ydir)
                os.remove(filename)

# Close FTP connection
f.close()